packages/toast/src/Toast.tsx

```diff
@@ -0,0 +1,335 @@
+import { Body } from '@leafygreen-ui/typography';
+import IconButton from '@leafygreen-ui/icon-button';
+import CheckmarkWithCircleIcon from '@leafygreen-ui/icon/dist/CheckmarkWithCircle';
+import RefreshIcon from '@leafygreen-ui/icon/dist/Refresh';
+import XIcon from '@leafygreen-ui/icon/dist/X';
+
+const progressBackgroundBase = '#22B7EB';
+const toastWidth = 400;
+
+const progressBarBackgroundStyle = css`
+  position: absolute;
+  bottom: 0;
+  left: 0;
+  right: 0;
+  height: 6px;
+  background-color: ${uiColors.gray.light2};
+  border-radius: 0 0 4px 4px;
+  overflow: hidden;
+`;
+
+const backgroundShimmer = keyframes`
+  0% {
+    background-position: ${-toastWidth}px;
+  }
+
+  100% {
+    background-position: ${toastWidth * 2}px;
+  }
+`;
+
+const progressBarStyle = css`
+  overflow: hidden;
+  height: 6px;
+  position: absolute;
+  top: 0;
+  bottom: 0;
+  left: 0;
+  background-color: ${progressBackgroundBase};
+  background-image: linear-gradient(
+    90deg,
+    ${progressBackgroundBase} 0px,
+    #cce8f4 ${toastWidth / 2}px,
+    ${progressBackgroundBase} ${toastWidth}px
+  );
+  background-size: 600px;
+  animation: ${backgroundShimmer} 4s infinite linear;
+  transition: width 0.3s ease-in-out;
+`;
+
+interface ProgressBarProps {
+  progress: number;
+}
+
+function ProgressBar({ progress }: ProgressBarProps) {
+  return (
+    <div className={progressBarBackgroundStyle}>
+      <div
+        className={cx(
+          progressBarStyle,
+          css`
+            width: ${progress * 100}%;
+          `,
+        )}
+      />
+    </div>
+  );
+}
+
+const RTGStates = {
+  Entering: 'entering',
+  Entered: 'entered',
+  Exiting: 'exiting',
+  Exited: 'exited',
+} as const;
+
+type RTGStates = typeof RTGStates[keyof typeof RTGStates];
+
+const ToastVariants = {
+  success: 'success',
+  progress: 'progress',
+} as const;
+
+type ToastVariants = typeof ToastVariants[keyof typeof ToastVariants];
+
+const transitionDuration = 150;
+
+const baseToastStyle = css`
+  position: fixed;
+  bottom: ${spacing[6]};
+  left: ${spacing[4]};
+  width: ${toastWidth}px;
+  max-width: calc(100vw - (${spacing[4]} * 2));
+  border-radius: 4px;
+  box-shadow: 0 18px 18px -15px ${transparentize(0.7, uiColors.black)};
+  overflow: hidden;
+  transform: translate3d(0, ${spacing[3]}, 0) scale(0.95);
+  transform-origin: bottom center;
+  opacity: 0;
+  transition: all ${transitionDuration}ms ease-in-out;
+`;
+
+const toastVariantStyles: Record<ToastVariants, string> = {
+  [ToastVariants.success]: css`
+    background-color: ${uiColors.green.light3};
+  `,
+
+  [ToastVariants.progress]: css`
+    background-color: ${uiColors.white};
+    padding-bottom: 6px;
+  `,
+};
+
+const toastVisibleStateStyle = css`
+  transform: translate3d(0, 0, 0) scale(1);
+  opacity: 1;
+`;
+
+const toastTransitionStateStyles: Partial<Record<RTGStates, string>> = {
+  [RTGStates.Entered]: toastVisibleStateStyle,
+};
+
+const baseIconStyle = css`
+  flex-shrink: 0;
```

Comment:
Did we bake this into Icon already? Or is this a part of a PR that hasn't merged yet 🤔

Reply:
Unsure – I did this to be safe TBH 😅 I don't think this was included quite yet though.